src/ApparentHorizons/YlmSpherepack.cpp

```diff
@@ -525,27 +524,31 @@ YlmSpherepack::first_and_second_derivative(
   return result;
 }
 
-YlmSpherepack::InterpolationInfoPerPoint::InterpolationInfoPerPoint(
+template <typename T>
+YlmSpherepack::InterpolationInfo<T>::InterpolationInfo(
     const size_t m_max, const std::vector<double>& pmm,
-    const std::array<double, 2>& target)
+    const std::array<T, 2>& target)
     : cos_theta(cos(target[0])),
       cos_m_phi(m_max + 1),
       sin_m_phi(m_max + 1),
-      pbar_factor(m_max + 1) {
-  const double sin_theta = sin(target[0]);
-  const double phi = target[1];
+      pbar_factor(m_max + 1),
+      num_points(get_size(target[0])) {
+  // do not use auto here, the blaze expression templates will create a dangling
```

Comment:
I spend the better part of a day debugging this. I thought I could be clever and delay the evaluation of the blaze expression templates essentially by using `auto` instead of `T` everywhere in these functions. This probably creates some sort of dangling reference as the expression templates are assigned. There is no warning or error during compile- or runtime, or even a segfault. The values are simply different so the unit test fails which is a complete nightmare to debug. Is this something that is known/expected or should we add some sort of cautionary documentation somewhere? @nilsdeppe

Reply:
@moxcodes I remember you had an issue that may or may not be related (#2728)

Reply:
Unfortunately, this is a problem that we have encountered a small handful of times, but the cases were spotty and in very complicated expressions, so I haven't taken the time to hunt through precisely which blaze calls are causing the bug. I'm pretty sure that this is a blaze bug or a compiler edge case in evaluating the complicated blaze expression templates, but I didn't have enough specific information about the problem to file a blaze issue.
If you think you have a clear self-contained case that shows the problem, then it would be good to open an issue with blaze. Otherwise, you could open an issue here on our github pointing out the case that you saw the segfault and then it will be easier for someone to follow up on it.